src/custom/controller.go

```diff
@@ -0,0 +1,175 @@
+	"k8s.io/client-go/util/workqueue"
+)
+
+const SecretMounter string = "secret-mounter"
+const DefaultContainerSecretPath string = "/etc/" + SecretMounter + "-data/"
+
+// Mandatory field
+const DeploymentLabelSecretName string = "secret-name"
+// Optional field
+const DeploymentLabelSecretKeys string = "secret-keys"
+// Secret Keys separator
+const SecretKeysSeparator string = "."
+
+type controller struct {
+	clientSet      kubernetes.Interface
+	depLister      appsListers.DeploymentLister
+	depCacheSynced cache.InformerSynced
+	workQueue      workqueue.RateLimitingInterface
+}
+
+func (cntrlReceiver *controller) handleAdd(obj interface{}) {
+	cntrlReceiver.workQueue.Add(obj)
+}
+
+func (cntrlReceiver *controller) Run(stopCh <-chan struct{}) {
+	fmt.Println("Starting Custom Controller....")
+	if !cache.WaitForCacheSync(stopCh, cntrlReceiver.depCacheSynced) {
+		fmt.Println("Waiting for the cache to be synced....")
+	}
+	go wait.Until(cntrlReceiver.worker, 1*time.Second, stopCh)
+	<-stopCh
+}
+
+func (cntrlReceiver *controller) worker() {
+	for cntrlReceiver.processItem() {
+	}
+}
+
+func (cntrlReceiver *controller) processItem() bool {
+	item, shutdown := cntrlReceiver.workQueue.Get()
+	if shutdown {
+		return false
+	}
+	defer cntrlReceiver.workQueue.Forget(item)
+	key, err := cache.MetaNamespaceKeyFunc(item)
+	if err != nil {
+		fmt.Println("Error in cache.MetaNamespaceKeyFunc(): ", err.Error())
+	}
+	ns, name, err := cache.SplitMetaNamespaceKey(key)
+	if err != nil {
+		fmt.Println("Error in cache.SplitMetaNamespaceKey(): ", err.Error())
+	}
+	err = cntrlReceiver.checkDeployments(ns, name)
+	if err != nil {
+		fmt.Println("Error in cntrlReceiver.checkDeployments(): ", err.Error())
+		return false
+	}
+	return true
+}
+
+func (cntrlReceiver *controller) checkDeployments(ns, name string) error {
+	// Get Deployment from api-server
+	ctx := context.Background()
+	deployment, err := cntrlReceiver.depLister.Deployments(ns).Get(name)
+	if err != nil {
+		fmt.Println("Error in cntrlReceiver.depLister.Deployments(ns).Get(): ", err.Error())
+	}
+	// Check the deployment metadata.labels
+	depLabels := deployment.Labels
+	if labelFilter, labelFilterOk := depLabels["app"]; labelFilterOk && labelFilter == SecretMounter {
+		if secretName, secretNameFilterOk := depLabels[DeploymentLabelSecretName]; secretNameFilterOk {
+			// Get the secret from the api-server
+			secret, err := cntrlReceiver.clientSet.CoreV1().Secrets(ns).Get(ctx, secretName, metaV1.GetOptions{})
+			if err != nil {
+				fmt.Println("Error in cntrlReceiver.clientSet.CoreV1().Secrets(ns).Get(): ", err.Error())
+				return err
+			}
+			// Mount secret as a volume in deployment
+			err = cntrlReceiver.mountSecretInDep(ctx, ns, name, *secret, depLabels)
+			if err != nil {
+				fmt.Println("Error in cntrlReceiver.mountSecretInDep(): ", err.Error())
+				return err
+			}
+		} else {
+			fmt.Println("Secret name not found in the deployment metadata.labels - Skipping secret mount....")
+		}
```

Comment:
else condition is not needed if we add return at the end of if statements
```suggestion
                        return nil
		}
		fmt.Println("Secret name not found in the deployment metadata.labels - Skipping secret mount....")

```